ssl/t1_trce.c

```diff
@@ -1493,7 +1497,13 @@ void SSL_trace(int write_p, int version, int content_type,
     switch (content_type) {
     case SSL3_RT_HEADER:
         {
-            int hvers = msg[1] << 8 | msg[2];
+            int hvers;
+            /* avoid overlapping with length at the end of buffer */
+            if (msglen < SSL_IS_DTLS(ssl) ? 13 : 5) {
+                ssl_print_hex(bio, 0, "too short message", msg, msglen);
+                break;
+            }
+            hvers = msg[1] << 8 | msg[2];
             BIO_puts(bio, write_p ? "Sent" : "Received");
```

Comment:
Direction is meaningful and presumably valuable even in "too short message" case. BTW, while you are at it, it's suggested to update coding style guide and say that declarations should be delimited by empty line. In other words could you add an empty line after "int hvers"? :-)